app/controllers/HomeController.scala

```diff
@@ -166,56 +166,54 @@ class HomeController @Inject()(val homeView: views.html.Home,
         case Left(ex: Throwable) =>
           Logger("application").error(s"Unable to create the view model ${ex.getMessage} - ${ex.getCause}")
           handleErrorGettingDueDates(user.isAgent())
+      }
     }
-}
   }
 
-  private def getDueDates(unpaidCharges: List[FinancialDetailsResponseModel]): List[LocalDate] =
-  (unpaidCharges collect {
-    case fdm: FinancialDetailsModel => fdm.validChargesWithRemainingToPay.getAllDueDates
-  })
-    .flatten
-    .sortWith(_ isBefore _)
-    .sortBy(_.toEpochDay())
-
-private def getOutstandingChargesModel(unpaidCharges: List[FinancialDetailsResponseModel])
-                                      (implicit user: MtdItUser[_]): Future[List[OutstandingChargeModel]] =
-  whatYouOweService.getWhatYouOweChargesList(
-    unpaidCharges,
-    isReviewAndReconciledEnabled = isEnabled(ReviewAndReconcilePoa),
-    isFilterCodedOutPoasEnabled = isEnabled(FilterCodedOutPoas)
-  ) map {
-    case WhatYouOweChargesList(_, _, Some(OutstandingChargesModel(outstandingCharges)), _) =>
-      outstandingCharges.filter(_.isBalancingChargeDebit)
-        .filter(_.relevantDueDate.isDefined)
-    case _ => Nil
-  }
+  private def getDueDates(unpaidCharges: Option[FinancialDetailsResponseModel]): List[LocalDate] =
+    (unpaidCharges collect {
+      case fdm: FinancialDetailsModel => fdm.validChargesWithRemainingToPay.getAllDueDates
+    }).getOrElse(List.empty)
```

Comment:
its looks like we are dropping sorting for some reason, unless this is part of the requirements for current story, its worth to double check